notebooks/project_demo.ipynb

```diff
@@ -0,0 +1,523 @@
+    "Additionally, GPT-4o can also make a lot of other mistakes. However, in this case, it worked well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "185182790a583c2b",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-15T12:20:18.772783Z",
+     "start_time": "2024-10-15T12:20:18.770174Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Original Value: \t230000000000.0\n",
+      "Extracted Value: \t250200000000\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Original Value: \\t{revenue_datapoint.value}\")\n",
+    "print(f\"Extracted Value: \\t{extracted_revenue}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d867964f5e18b392",
+   "metadata": {},
+   "source": "We can directly see that the values do not align. Therefore, the claim is incorrect. We report this information back to the data provider by creating a so-called QA Report"
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8850c66f1c4439d7",
+   "metadata": {},
+   "source": "# 4. Creating and submitting a QA Report"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "37b44332f780e102",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-15T12:20:23.728206Z",
+     "start_time": "2024-10-15T12:20:22.723009Z"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "QaReportMetaInformation(data_id='16c10aa8-689c-4773-919e-bb493b700db5', data_type='eutaxonomy-non-financials', qa_report_id='ac544179-f0de-4c1b-99c1-69992d4da295', reporter_user_id='9697edcb-dab7-41f1-aa69-97338e7300fe', upload_time=1728994823481, active=True)"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from dataland_qa.models.eutaxonomy_non_financials_data import EutaxonomyNonFinancialsData\n",
+    "from dataland_qa.models.qa_report_data_point_verdict import QaReportDataPointVerdict\n",
+    "\n",
+    "selected_qa_report = EutaxonomyNonFinancialsData.from_dict(\n",
+    "    {\n",
+    "        \"revenue\": {\n",
+    "            \"totalAmount\": {\n",
+    "                \"comment\": \"The total revenue is incorrect. The correct value is 250200000000 €\",\n",
+    "                \"verdict\": QaReportDataPointVerdict.QAREJECTED,\n",
+    "                \"correctedData\": {\n",
+    "                    \"value\": extracted_revenue,\n",
+    "                    \"quality\": \"Reported\",\n",
+    "                    \"comment\": parsed_tool_arguments[\"justification\"],\n",
+    "                    \"currency\": parsed_tool_arguments[\"answer_currency\"],\n",
+    "                    \"dataSource\": revenue_datapoint.data_source.model_dump(by_alias=True),\n",
+    "                },\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    ")\n",
```

Comment:
Instead of using from dict and a magic string let's use Dataland objects to make the students more aware of their existence:

`selected_qa_report = EutaxonomyNonFinancialsData(
	revenue=EutaxonomyNonFinancialsRevenue(
		totalAmount=QaReportDataPointCurrencyDataPoint(
			comment="The total revenue is incorrect. The correct value is 250200000000 €",
			verdict=QaReportDataPointVerdict.QAREJECTED,
			correctedData=CurrencyDataPoint(
				value=extracted_revenue,
				quality="Reported",
				comment=parsed_tool_arguments["justification"],
				currency=parsed_tool_arguments["answer_currency"],
                dataSource=ExtendedDocumentReference.from_dict(revenue_datapoint.data_source.model_dump(by_alias=True))
			)
		)
	)
)`

Reply:
Fixed